datafusion/sql/src/expr/mod.rs

```diff
@@ -341,7 +278,17 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                     }
                 };
 
-                self.plan_field_access(expr, get_field_access)
+                let mut field_access_expr = RawFieldAccessExpr { expr, field_access };
+                for planner in self.planners.iter() {
+                    match planner.plan_field_access(field_access_expr, schema)? {
+                        PlannerResult::Planned(expr) => return Ok(expr),
+                        PlannerResult::Original(expr) => {
+                            field_access_expr = expr;
+                        }
+                    }
+                }
+
+                not_impl_err!("GetFieldAccess not supported by UserDefinedExtensionPlanners: {field_access_expr:?}")
```

Comment:
:wave: I'm curious about this default branch because I'm seeing this error when doing integration testing with the delta-rs package.  To the best of my knowledge we're not using any extension planners, but now [this is failing](https://github.com/delta-io/delta-rs/blob/main/crates/core/src/operations/delete.rs#L769-L773) with the latest datafusion.  I'm uncertain whether we're doing something wrong, or if the default behavior of falling through to `not_impl_err` is causing us grief

Reply:
Hi @rtyler  -- looks to me like the delta.rs code is managing its own SessionState -- like in https://github.com/delta-io/delta-rs/blob/main/crates/core/src/delta_datafusion/mod.rs#L1687 (BTW the Delta API is really nicely thought out)

So I think you'll need to register the same planners in your SessionContext 🤔 

Helpfully I think @Omega359  just made a PR to make this easier: https://github.com/apache/datafusion/pull/11296

I feel in general DataFusion is hard to use / configure correctly if you are using a custom SessionState / configuration -- one potential thing we were discussing is https://github.com/apache/datafusion/issues/11182#issuecomment-2200327993 -- I'll file a ticket to make this a thing.

Reply:
I think I had a similar issue you you @rtyler. Found a workaround described here: https://github.com/apache/datafusion/issues/11477

Reply:
Proposed fix is here: https://github.com/apache/datafusion/pull/11485